### PR TITLE
RHOAIENG-15772: tests(odh-nbc): wait for controller manager to stop before stopping envtest

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-15772

Relates to
* https://github.com/opendatahub-io/kubeflow/issues/429

## Description

Otherwise, there will be some errors printed in the output. It was not breaking anything, but it was unpleasant to look at.

The notebook-controller does not seem to be suffering from this. Probably because it does not use a webhook.

Example of the error message that's prevented by this

```
2024-11-17T20:19:57+01:00       ERROR   controller-runtime.certwatcher  error re-reading certificate    {"error": "open /var/folders/f1/3m518k5d34l72v_9nqyjzqm80000gn/T/envtest-serving-certs-737480808/tls.crt: no such file or directory"}
```

## How Has This Been Tested?

```
make test
```

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/odh-notebook-controller/controllers/suite_test.go [hunk 1]
```diff
@@ -55,11 +55,14 @@ import (
 // +kubebuilder:docs-gen:collapse=Imports
 
 var (
-	cfg            *rest.Config
-	cli            client.Client
-	envTest        *envtest.Environment
+	cfg     *rest.Config
```
Comment (jiridanek) [inline components/odh-notebook-controller/controllers/suite_test.go:58]: The linter is right, but that's issue that has been in the repo for a long time already. See here for fix
* https://github.com/opendatahub-io/kubeflow/pull/462

**Comment by jstourac** [comment]
/lgtm

Thank you for this!

**Comment by jiridanek** [comment]
/approve
the failing test should go away when prow retests, or maybe I'd need to retrigger manually

File: components/odh-notebook-controller/controllers/suite_test.go (hunk 1 above)
Comment (jiridanek) [reply components/odh-notebook-controller/controllers/suite_test.go:58]: /override "Code static analysis / golangci-lint (components/odh-notebook-controller)"

Comment (jiridanek) [reply components/odh-notebook-controller/controllers/suite_test.go:58]: /override "Code static analysis / golangci-lint (components/odh-notebook-controller)"

Comment (jiridanek) [reply components/odh-notebook-controller/controllers/suite_test.go:58]: /override "golangci-lint (components/odh-notebook-controller)"

**Comment by jstourac** [comment]
> /approve the failing test should go away when prow retests, or maybe I'd need to retrigger manually

shouldn't you rebase first to make this work? or we can just override?

**Comment by jiridanek** [comment]
I guess I'll have to do that. My reasoning was that I did not want to lose the /lgt,

**Comment by jiridanek** [comment]
wow, the update of a PR branch with merge really shows much more cleanly in GH ui; but I'm still partial to rebases, old habits

File: components/odh-notebook-controller/controllers/suite_test.go (hunk 1 above)
Comment (jiridanek) [reply components/odh-notebook-controller/controllers/suite_test.go:59]: yeah, I <s>rebased</s> merged

**Pull request merged**